### PR TITLE
dynamic_test: Replace goroutine with time.After()

### PR DESCRIPTION
**Pull request by gaurav1086**
Signed-off-by: Gaurav Singh <gaurav1086@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Minor cleanup: Replace goroutine with time.After()
**Which issue(s) this PR fixes**:

Fixes #
N/A
**Special notes for your reviewer**:
Minor cleanup.
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
/ok-to-test

/lgtm

File: staging/src/k8s.io/apiserver/plugin/pkg/audit/dynamic/dynamic_test.go [hunk 1]
```diff
@@ -233,20 +233,15 @@ func TestDynamic(t *testing.T) {
...
-		go func() {
-			time.Sleep(1 * time.Second)
-			timeoutChan <- struct{}{}
-		}()
 		go func() {
 			close(stopChan)
 			d.Shutdown()
 			successChan <- struct{}{}
 		}()
 		for {
```
Comment (MHBauer) [inline staging/src/k8s.io/apiserver/plugin/pkg/audit/dynamic/dynamic_test.go:242]: what does the for loop do if both paths return from the Run func()?

**Review by MHBauer — APPROVED** [review]
/lgtm

File: staging/src/k8s.io/apiserver/plugin/pkg/audit/dynamic/dynamic_test.go (hunk 1 above)
Comment (gaurav1086) [reply staging/src/k8s.io/apiserver/plugin/pkg/audit/dynamic/dynamic_test.go:242]: @MHBauer yes, the for loop is unnecessary. I think we can remove it.

**Comment by lavalamp** [comment]
We're about to delete all the dynamic audit stuff. @deads2k let us know if this code won't be deleted?

**Comment by cheftako** [comment]
Dynamic has been deleted at this point.

**Comment by lavalamp** [comment]
Yeah and this has a merge conflict now. The best fix is always deleting stuff ;)

**Pull request closed without merging**